lib/experimental/block-bindings/class-wp-block-bindings.php

```diff
@@ -107,4 +155,22 @@ function block_bindings_replace_html( $block_content, $block_name, $block_attr,
 		}
 		return;
 	}
+
+	/**
+	 * Retrieves the list of registered block sources.
+	 *
+	 * @return array The array of registered sources.
+	 */
+	public function get_sources() {
+		return $this->sources;
+	}
+
+	/**
+	 * Retrieves the list of allowed blocks that support block bindings.
+	 *
+	 * @return array The array of allowed blocks.
+	 */
+	public function get_allowed_blocks() {
+		return $this->allowed_blocks;
+	}
```

Comment:
Just thinking about a future where this mechanism may change (it becomes an opt-in or opt-out, or we support so many core blocks that it becomes easier to list the ones that aren't supported), and ways that the method could be future proofed.

Maybe changing it to something like `get_is_block_type_supported( $block_type )` could be a way to future proof it. That would prevent leaking the `allowed_blocks` array into public code and allow us to change the way it's implemented while keeping the interface the same.

Reply:
I definitely agree with the rationale. What do you think about going even further and making it an implementation detail of the function that takes care of computing the value of the attributes that have a corresponding custom source binding? If there are no attribute changes then there is nothing much to process in the block content's HTML.

Reply:
I think what Dan suggests makes total sense. The `allowed_blocks` property is already `private` so if we use a function like `get_is_block_type_supported( $block_type )` instead it should be enough to "future proof" it.

I'm not sure I follow your suggestion @gziolo though. Could you elaborate?

Reply:
If we go with this, we should also update the wrapper function in: https://github.com/WordPress/gutenberg/blob/7252c01c54df85de3dfea320b336bbcb9751dc72/lib/experimental/block-bindings/block-bindings.php#L50-L54

Reply:
Thanks Greg! Yeah, that makes sense to me. 👍 

There's no need to expose `wp_block_bindings_get_allowed_blocks()` as public API so we can remove it as well as remove the internal `$this->get_allowed_blocks()` method.

Reply:
Ok I moved the $allowed_blocks property to the filter function and removed the loop 👍
Perhaps the motivation of the original code was to prevent running the filter function on every block on the page?

In any case, I think what I've [revised here](https://github.com/WordPress/gutenberg/pull/57742/commits/d7e1c93bd852bd754561d7f6d7e6a2fba92c01dd) addresses the issue. Let me know what you think!